doc/maintenance.md

```diff
@@ -79,8 +79,7 @@ There are 11 configuration files located in `.github/workflows`:
 2. `ci_tests.yaml` (Tests on Linux/macOS/Windows)
```

Comment:
**Off-topic question**: Currently, we're documenting what a workflow does in the maintainers' guide, but it's very likely that we may forget to update these descriptions when we update a workflow. Perhaps we should only keep a short, one-sentence description for each workflow here and have more detailed descriptions in the workflow file.

Reply:
Maybe have a README.md under https://github.com/GenericMappingTools/pygmt/tree/v0.9.0/.github/workflows to document, so that people looking at the folder can see what the *.yml files are for? Also fine with including it in the YAML file since that will be more visible.

Reply:
> Also fine with including it in the YAML file since that will be more visible.

I'm inclined to this option because (1) we won't forget to update the workflow descriptions; (2) the workflows are more self-explanatory.

Reply:
> > Also fine with including it in the YAML file since that will be more visible.
> 
> I'm inclined to this option because (1) we won't forget to update the workflow descriptions; (2) the workflows are more self-explanatory.

Ok, so just document things in the YAML files and remove the descriptions from the doc/maintenance.md file. Can do this in another PR once this one is merged.